src/sidebar/test/session-test.js

```diff
@@ -10,6 +10,7 @@ describe('session', function () {
   var $httpBackend;
   var $rootScope;
 
+  var fakeAnalytics;
```

Comment:
Could you add a couple of tests here to check that the expected events are tracked for login/logout success and failure?

Related to this, is there any way we could easily do "testing in production" by getting alerts if there was a sudden change to metrics - eg. say for some reason the client stopped sending analytics events due to misconfiguration?